tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -730,7 +730,20 @@ object CodeGen {
           ()
 
         case call: Op.Call =>
-          genCall(genBind, call, unwind)
+          val withGeneratedExtern = call match {
+            case Op.Call(ty, Val.Global(g @ Global.Member(_, sig), valty), args)
+                if sig.isExtern =>
+              deps.collectFirst {
+                case gg @ Global.Member(_, `sig`) => gg
+              } match {
+                case Some(gg) if gg != g =>
+                  Op.Call(ty, Val.Global(gg, valty), args)
+                case _ => call
+              }
+            case _ =>
+              call
+          }
+          genCall(genBind, withGeneratedExtern, unwind)
```

Comment:
I don't really like idea of traversing `deps` each time. You may consider adding `val externSigMembers = mutable.Map.empty[Sig, Global.Member]` and use it instead of `deps` traversal. Also I would consider it as hacking so we should provide some comment describing it. 

```suggestion
         /*In case if it's a call to extern method with same mangled Sig as some already defined call
            in other module we need to manually enforce getting into second case of `genCall`
            (when lookup(pointee) != call.ty). By replacing `call.ptr` with ptr of that already defined call we would
            enforce creating call bitcasts to correct type.

            Due to deduplication in `genDeps` and since mangling Sig.Extern is not based on function types
            each extern method in deps is generated only once in IR file. In this case LLVM linking
            would result in call arguments type mismatch.
          */
          val callDef = call.ptr match {
            case Val.Global(m @ Global.Member(_, sig), valty) if sig.isExtern =>
              val glob = externSigMembers.getOrElseUpdate(sig, m)
              if (glob == m) call
              else call.copy(ptr = Val.Global(glob, valty))
            case _ => call
          }
          genCall(genBind, callDef, unwind)
```

Reply:
Thank you for the suggestion @WojciechMazur. I'm going to apply it as soon as I can.